viadot/sources/mindful.py

```diff
@@ -0,0 +1,234 @@
+            params (Dict[str, Any], optional): Parameters to be passed into the request. Defaults to None.
+            endpoint (str, optional): API endpoint for an individual request. Defaults to "".
+
+        Returns:
+            Response: request object with the response from the Mindful API.
+        """
+
+        response = handle_api_response(
+            url=f"https://{self.region}surveydynamix.com/api/{endpoint}",
+            params=params,
+            headers=self.header,
+            method="GET",
+        )
+
+        return response
+
+    def get_interactions_list(
+        self,
+        limit: int = 1000,
+        **kwargs,
+    ) -> Response:
+        """Gets a list of survey interactions as a JSON array of interaction resources.
+
+        Args:
+            limit (int, optional): The number of matching interactions to return. Defaults to 1000.
+
+        Returns:
+            Response: request object with the response from the Mindful API.
+        """
+
+        self.endpoint = "interactions"
+        params = {
+            "_limit": limit,
+            "start_date": f"{self.start_date}",
+            "end_date": f"{self.end_date}",
+        }
+
+        response = self._mindful_api_response(
+            endpoint=self.endpoint,
+            params=params,
+        )
+
+        if response.status_code == 200:
+            self.logger.info(
+                "Succesfully downloaded interactions data from mindful API."
+            )
+        else:
+            self.logger.error(
+                f"Failed to downloaded interactions data. - {response.content}"
+            )
+            raise APIError("Failed to downloaded interactions data.")
+
+        return response
+
+    def get_responses_list(
+        self,
+        limit: int = 1000,
+        **kwargs,
+    ) -> Response:
+        """Gets a list of survey responses associated with a survey, question, or interaction resource.
+
+        Args:
+            limit (int, optional): The number of matching interactions to return. Defaults to 1000.
+
+        Returns:
+            Response: request object with the response from the Mindful API.
+        """
+
+        self.endpoint = "responses"
+        params = {
+            "_limit": limit,
+            "start_date": f"{self.start_date}",
+            "end_date": f"{self.end_date}",
+        }
+
+        response = self._mindful_api_response(
+            endpoint=self.endpoint,
+            params=params,
+        )
+
+        if response.status_code == 200:
+            self.logger.info("Succesfully downloaded responses data from mindful API.")
+        else:
+            self.logger.error(
+                f"Failed to downloaded responses data. - {response.content}"
+            )
+            raise APIError("Failed to downloaded responses data.")
+
+        return response
+
+    def response_to_file(
+        self,
+        response: Response,
+        file_name: str = None,
+        file_path: str = "",
```

Comment:
I understand your reason, but file_name is not an option in the task, because it generates more than one. But the path can be different.

but with that comment, I realize about a little bug in the code ;) that I'm going to solve right now